### DISCUSSION TITLE
limit amount for useable homes

### DISCUSSION THREAD

**Discussion by CloudeLecaw** [Zrips/CMI#7825] [Feature Requests]
we currently can allow different groups to set different amount of homes

**Scenario**:
Player joins the server and has access to create 3 Homes. he create his 3 homes and can use all 3.
Now player decides to become VIP what is a limited timerank of 30 days.
in this rank he is allowed to set 10 homes.
he sets all 10  homes and can use all 10..
The month is over and the player becomes a normal player aggain.

--------------------------------------------------------------

**Issue now**: 
He cant create new homes because he is limited to 3 homes, what is fine and correct.
BUT he still can use all 10 homes he created before as VIP, even having 10 homes should be the VIP feature in this case. So if he dont care about setting new homes because he got all homepoint he needs with the 10 he made as VIP. Then he has access to this 10 points forever...

--------------------------------------------------------------

**Idea 1**: adding a limitation permission for homes

so it would be nice to have a per permission limitation like **_cmi.commamd.home.[amount]_**

if the value is for example **_cmi.command.home.3_**
he only can use the first 3 homes in his list.

if he would regain VIP the permission would change to **_cmi.commamd.home.10_** what would allowed him to use all 10 aggain.

like for for sethome a unlimited permission would be needed too for admins may something like
**_cmi.commamd.home.unlimited_**

--------------------------------------------------------------

**Idea 2**: limited useage depending on sethome permission

The home useage could be limited with the allready existing permission **_cmi.commamd.sethome.[amount]_** permission.
in this case the permission would limit the amount of homes that can be set and be used.
so if player would have.
**_cmi.commamd.sethome.3_** but created 10 homes because he had **_cmi.commamd.sethome.10_** before, he would only be able to use the first 3 homes from his list.

--------------------------------------------------------------

might it also needs a little line like:
 "_You cant use this homepoint, because its over your allowed limit._"
if you try to acces a homepoint that is higher than the home permission you got.

**Comment by MrCrazys** [comment]
This would likely be hard to implement, and how would the player be able to decide which homes they want to be able to teleport to still? I imagine this would be very taxing, as there would likely be a lot of checks involved.

**Reply by CloudeLecaw** [reply]
> This would likely be hard to implement, and how would the player be able to decide which homes they want to be able to teleport to still? I imagine this would be very taxing, as there would likely be a lot of checks involved.

they DONT have to decide.
They have 3 normaly the first 3 in our case.. when they are VIP then can create 7 more the first 3 are still the first 3 in list.. if they are in nrmal Rank than they only have this 3 anymore. Its not like Player is allowed to choose "ohh i want this or that from the 10" Its the first 3 he made.  Also he has athe option to remove the other  homes each time if he wants one of the  7 he dont has acces to anymore..

**Comment by TomLewis** [comment]
I have this same issue.

Just being able to use their first X homes is the simplest option.
These are always usable, and they keep their other ones set at a higher rank, so if they get the rank again they won't have lost the homes they set.

It's also what users would expect it to work. 

So it defaults back to which permission they still have, it just makes the existing ones they have set not usable. No need for extra permissions.

**Reply by CloudeLecaw** [reply]
> I have this same issue.
> 
> Just being able to use their first X homes is the simplest option. These are always usable, and they keep their other ones set at a higher rank, so if they get the rank again they won't have lost the homes they set.
> 
> It's also what users would expect it to work.
> 
> So it defaults back to which permission they still have, it just makes the existing ones they have set not usable. No need for extra permissions.

yes they should not loose just. just have no acces anymore. They should not get deleted just no access to use or simply just not able to see/use them.
Removing them would not be good because then if they regain VIP you can aspect they may wanne use their old homes aggain.

**Reply by MrCrazys** [reply]
My apologies, I misunderstood you. I agree, it would be good if the homes were still there, just not accessible.
